--- dictra_analyzr/corrector.py
+++ dictra_analyzr/corrector.py
@@ -148,8 +148,10 @@

     def _get_new_phase_name(self, phase_to_change, sorted_elnames, cutoff):
-        if 0 < cutoff < 1:
+        if cutoff == 1:
+             return phase_to_change + '-' + sorted_elnames[0]
+        elif cutoff > 0 and cutoff < 1:
              return phase_to_change + '-' + "".join(sorted_elnames[:2])
         else:
-            return phase_to_change + '-' + "".join(sorted_elnames[:1])
+             return phase_to_change + '-' + "".join(sorted_elnames)

     def _cleanup_empty_phases(self, d):
